File: model/src/test/java/com/lightbend/model/SimpleTensorFlowBundleModel.java
package com.lightbend.model;

import com.lightbend.model.tensorflow.TensorFlowBundleModel;
import org.tensorflow.Graph;
import org.tensorflow.Session;

import java.util.List;
import java.util.Map;

// Tensorflow bundle implementation for testing
public class SimpleTensorFlowBundleModel extends TensorFlowBundleModel {

    public SimpleTensorFlowBundleModel(byte[] inputStream)  {
        super(inputStream);
    }

    @Override
    public Object score(Object input) {
        // Just for test
        return null;
    }

    // Getters for testing

    public Graph getGraph() {
        return graph;
    }

    public Session getSession() {
        return session;
    }

    public Map<String, Signature> getSignatures() {
        return signatures;
    }

    public List<String> getTags() {
        return tags;
    }
}

File: model/src/test/scala/com/lighbend/modelServer/SimpleTensorflowBundleModel.scala
package com.lighbend.modelServer


import com.lightbend.modelServer.ModelToServe
import com.lightbend.modelServer.model.{Model, ModelFactory}
import com.lightbend.modelServer.model.tensorflow.{Signature, TensorFlowBundleModel}
import org.tensorflow.{Graph, Session}

class SimpleTensorflowBundleModel(inputStream: Array[Byte]) extends TensorFlowBundleModel(inputStream) {

  override def score(input: AnyVal): AnyVal = null.asInstanceOf[AnyVal]

  // Getters for testing
  def getGraph: Graph = graph

  def getSession: Session = session

  def getSignatures: Map[String, Signature] = parsedSign

  def getTags: Seq[String] = tags
}

object SimpleTensorflowBundleModel extends  ModelFactory {

  override def create(input: ModelToServe): Option[Model] = {
    try {
      Some(new SimpleTensorflowBundleModel(input.location.getBytes()))
    }catch{
      case t: Throwable => None
    }
  }

  override def restore(bytes: Array[Byte]): Model = new SimpleTensorflowBundleModel(bytes)
}
